docs/sisl/waterworld.md

```diff
@@ -1,71 +1,107 @@
 
-By default there are 5 agents (purple), 5 food targets (green) and 10 poison targets (red). Each agent has 30 range-limited sensors, depicted by the black lines, to detect neighboring agents (food and poison targets) resulting in 212 long vector of computed values about the environment for the observation space. They have a continuous action space represented as a 2 element vector, which corresponds to left/right and up/down thrust. The agents each receive a reward of 10 when more than one agent captures food together (the food is not destroyed), a shaping reward of 0.01 for touching food, a reward of -1 for touching poison, and a -0.5×｜action｜ reward when an agent collides into another. The environment runs for 500 frames by default. Observation shape takes the full form of ((4 + 3×speed_features)×n_sensors+2,).
+For example, by default there are 5 agents (purple), 5 food targets (green) and 10 poison targets (red). Each agent has 30 range-limited sensors, depicted by the black lines, to detect neighboring agents (food and poison targets) resulting in 242 long vector of computed values about the environment for the observation space. These values represent the distances and speeds sensed by each sensor on the archea. Sensors that do not sense any objects within their range report 0 for speed and 1 for distance.
 
 This has been fixed from the reference environments to keep items floating off screen and being lost forever.
 
-### Arguments
+This table enumerates the observation space with `speed_features = True`:
 
-```
-waterworld.env(n_pursuers=5, n_evaders=5, n_coop=2, n_poison=10, radius=0.015,
-obstacle_radius=0.2, obstacle_loc=np.array([0.5, 0.5]), ev_speed=0.01,
-poison_speed=0.01, n_sensors=30, sensor_range=0.2, action_scale=0.01,
-poison_reward=-1., food_reward=10., encounter_reward=.01, control_penalty=-.5,
-local_ratio=1.0, speed_features=True, max_cycles=500)
-```
+|        Index: [start, end)         | Description                                  | Values  |
+| :--------------------------------: | -------------------------------------------- | :-----: |
+|           0 to n_sensors           | Obstacle distance for each sensor            | [0, 1]  |
+|    n_sensors to (2 * n_sensors)    | Barrier distance for each sensor             | [0, 1]  |
+| (2 * n_sensors) to (3 * n_sensors) | Evader distance for each sensor              | [0, 1]  |
+| (3 * n_sensors) to (4 * n_sensors) | Evader speed for each sensor                 | [-1, 1] |
```

Comment:
It's sort of a side effect of the system rather than an actual hard limit. All objects have set speeds except for pursuers. If a pursuer ever has a speed greater than 1 or less than -1, then it would instantly run into a barrier at x=0, y=0, x=1, or y=1. Whenever a pursuer hits a wall, its velocity is set to 0. Feel free to correct me on any of that, but I think clipping makes it impossible to observe a speed over 1.

Also sensor speeds are absolute not relative, so we don't get situations where two pursuers moving in opposite directions would observe speeds over 1 or anything like that.

Reply:
shouldnt sensor speeds be relative?

Reply:
Never mind I forgot how that was coded, speeds are relative. I think that it's still impossible to observe a speed higher than 1. If two pursuers are moving in opposite directions from the center, the fastest they can move without hitting the edge is 0.5. If both move at that speed, they still combine to 1. But I guess I have to think about that a bit more.

Reply:
What if both start at opposite ends of the map and move past each other? Then can't they get to a speed of 2?

Reply:
Good call. I thought that the sensing range would prevent that but if you have two sensors pointing in opposite directions, one will pick up that really high speed. After doing some testing I was actually able to set them up and get a speed of 2.5 which is really surprising, so I'm going to look into this more closely.

Reply:
Okay, thanks for pointing this out, turns out position clipping doesn't actually work. I'll fix that now.